### PR TITLE
Operator Overloading Committed

### PR DESCRIPTION
**Pull request by tavisca-ishan**
Operator Overloading Review

### CODE REVIEW

File: 01 Operator Overloading/Operator Overloading.Model/Money.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    public class Money
+    {
+
+        private string _currency; //recieves currency
+        private double _amount; //recieves amount
+        double number;
+        public Money(string newString)
+        {
+            string[] split = newString.Split(' ');
```
Comment (tavisca-ishan) [inline 01 Operator Overloading/Operator Overloading.Model/Money.cs]: What will happen if `newString`?

File: 01 Operator Overloading/Operator Overloading.Model/Money.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  
+
+
+    public class Money
+    {
+
+        private string _currency; //recieves currency
+        private double _amount; //recieves amount
+        double number;
```
Comment (tavisca-ishan) [inline 01 Operator Overloading/Operator Overloading.Model/Money.cs]: what is this?

File: 01 Operator Overloading/Operator Overloading.Model/Money.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        public Money(string newString)
+        {
+            string[] split = newString.Split(' ');
+            if (split.Length != 2)
+            {
+                throw new System.Exception(ExceptionMessages.InvalidFormat);
+            }
+            if ((Double.TryParse(split[0], out number) != true))
```
Comment (tavisca-ishan) [inline 01 Operator Overloading/Operator Overloading.Model/Money.cs]: User `double`

File: 01 Operator Overloading/Operator Overloading.Model/Money.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                throw new System.Exception(ExceptionMessages.InvalidFormat);
+            }
+            else 
+            {
+                Amount = number;
+                Currency = split[1];
+            }
+        }
+
+        public Money(string currency, double amount)
```
Comment (tavisca-ishan) [inline 01 Operator Overloading/Operator Overloading.Model/Money.cs]: Swap parameters

File: 01 Operator Overloading/Operator Overloading.Model/Money.cs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            get
+            {
+                return _amount;
+            }
+
+
+        }
+
+        //overload + operator to add two money objects
+        public static Money operator +(Money first, Money second) //operator overloading
```
Comment (tavisca-ishan) [inline 01 Operator Overloading/Operator Overloading.Model/Money.cs]: Validate parameters. money1 and money2

File: 01 Operator Overloading/Operator Overloading.Model/Money.cs [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            }
+        }
+
+
+        public double Amount
+        {
+            private set
+            {
+
+                if (value <= 0 || value > Double.MaxValue)
```
Comment (tavisca-ishan) [inline 01 Operator Overloading/Operator Overloading.Model/Money.cs]: Use `double`

**Comment by tavisca-ishan** [comment]
@amarpalsapure please review operator overloading final commit

File: 01 Operator Overloading/Operator Overloading.Model/Money.cs [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            {
+                throw new System.Exception(ExceptionMessages.InvalidFormat);
+            }
+
+            double amount;
+            if ((double.TryParse(split[0], out amount) == false))
+            {
+                throw new System.Exception(ExceptionMessages.InvalidFormat);
+            }
+            else
```
Comment (apalsapure) [inline 01 Operator Overloading/Operator Overloading.Model/Money.cs]: This `else` part is not required.

**Comment by tavisca-ishan** [comment]
```
  @apalsapure please review the final amendments
```

**Pull request merged**